test/plots/barPlotTests.ts

```diff
@@ -1 +1,783 @@
+            });
+
+            afterEach(function() {
+              if (this.currentTest.state === "passed") {
+                barPlot.destroy();
+                svg.remove();
+              }
+            });
+          });
+        });
+      });
+
+      describe(`labels when ${orientation}`, () => {
+        const data = [
+          { base: -4, value: -4 },
+          { base: -2, value: -0.1},
+          { base: 0, value: 0 },
+          { base: 2, value: 0.1 },
+          { base: 4, value: 4 }
+        ];
+        const DEFAULT_DOMAIN = [-5, 5];
+
+        let svg: d3.Selection<void>;
+        let baseScale: Plottable.Scales.Linear;
+        let valueScale: Plottable.Scales.Linear;
+        let barPlot: Plottable.Plots.Bar<number, number>;
+        let dataset: Plottable.Dataset;
+
+        beforeEach(() => {
+          svg = TestMethods.generateSVG();
+          barPlot = new Plottable.Plots.Bar<number, number>(orientation);
+          baseScale = new Plottable.Scales.Linear();
+          baseScale.domain(DEFAULT_DOMAIN);
+          valueScale = new Plottable.Scales.Linear();
+          valueScale.domain(DEFAULT_DOMAIN);
+          if (orientation === Plottable.Plots.Bar.ORIENTATION_VERTICAL) {
+            barPlot.x((d: any) => d.base, baseScale);
+            barPlot.y((d: any) => d.value, valueScale);
+          } else {
+            barPlot.y((d: any) => d.base, baseScale);
+            barPlot.x((d: any) => d.value, valueScale);
+          }
+          dataset = new Plottable.Dataset(data);
+          barPlot.addDataset(dataset);
+          barPlot.renderTo(svg);
+        });
+
+        function getCenterOfText(textNode: SVGElement) {
+          const plotBoundingClientRect = (<SVGElement> barPlot.background().node()).getBoundingClientRect();
+          const labelBoundingClientRect = textNode.getBoundingClientRect();
+
+          return {
+            x: (labelBoundingClientRect.left + labelBoundingClientRect.right) / 2 - plotBoundingClientRect.left,
+            y: (labelBoundingClientRect.top + labelBoundingClientRect.bottom) / 2 - plotBoundingClientRect.top
+          };
+        };
+
+        it("does not show labels by default", () => {
+          const texts = barPlot.content().selectAll("text");
+          assert.strictEqual(texts.size(), 0, "by default, no texts are drawn");
+        });
+
+        it("draws one label per datum", () => {
+          barPlot.labelsEnabled(true);
+          barPlot.labelsEnabled(true);
+          const texts = barPlot.content().selectAll("text");
+          assert.strictEqual(texts.size(), data.length, "one label drawn per datum");
+          texts.each(function(d, i) {
+            assert.strictEqual(d3.select(this).text(), data[i].value.toString(), `by default, label text is the bar's value (index ${i})`);
+          });
+        });
+
+        it("hides the labels if bars are too thin to show them", () => {
+          svg.attr(baseSizeAttr, TestMethods.numAttr(svg, baseSizeAttr) / 10);
```

Comment:
Let's extract the second argument to a var to show that we are forcible constraining the space